crypto/src/crypto/agreement/ECVKOAgreement.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Math;
+using Org.BouncyCastle.Math.EC;
+using Org.BouncyCastle.Utilities;
+using System;
+
+namespace Org.BouncyCastle.Crypto.Agreement
+{
+    /// <summary>
+    /// GOST VKO key agreement class - RFC 7836 Section 4.3 
+    /// </summary>
+    public class ECVKOAgreement
+    {
+        private readonly IDigest digest;
+
+        private ECPrivateKeyParameters key;
+        private BigInteger ukm;
+
+        public int AgreementSize => this.digest.GetDigestSize();
+
+        public ECVKOAgreement(IDigest digest)
+        {
+            this.digest = digest;
+        }
+
+        public void Init(ICipherParameters parameters)
+        {
+            if (!(parameters is ParametersWithUKM paramsWithUkm))
+                throw new ArgumentException("ECVKOAgreement expects ParametersWithUKM");
+
+            if (!(paramsWithUkm.GetParameters() is ECPrivateKeyParameters ecParams))
+                throw new ArgumentException("ECVKOAgreement expects ParametersWithUKM contains ECPrivateKeyParameters");
+
+            this.key = ecParams;
+            this.ukm = new BigInteger(1, Arrays.Reverse(paramsWithUkm.GetUKM()));
```

Comment:
bc-csharp has BigInteger constructors with a "bool bigEndian" parameter if that is useful here.